src/api/app/controllers/public_controller.rb

```diff
@@ -81,6 +81,12 @@ def project_index
       # path has multiple package= parameters
       path += '?' + request.query_string
       path += '&nofilename=1' unless params[:nofilename]
+    elsif params['view'] == 'productlist'
+      render xml: @project.render_productlist(params[:expand])
```

Comment:
This shouldn't use a method in the Project model but an XML view like we use for [binary package list](https://github.com/openSUSE/open-build-service/blob/master/src/api/app/views/public/binary_packages.xml.builder) etc.